tests/main/snapshot-exclusions-dynamic/task.yaml

```diff
@@ -0,0 +1,132 @@
+        touch "$DIR/dynamic-exclude.txt"
+    done
+
+    # Create snapshot that will apply static exclusions, and grab the set ID
+    SET_ID=$( snap save test-snap | cut -d\  -f1 | tail -n1 )
+    echo "$SET_ID" | MATCH "^[0-9]+$"
+    
+    # Confirm the snapshot for the set ID includes the test-snap
+    snap saved --id="$SET_ID" | MATCH "test-snap"
+    # and is valid
+    snap check-snapshot "$SET_ID"
+    # and the snapshot metadata does not include an options entry
+    curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snapshots?set="$SET_ID" | NOMATCH '"options":'
+
+    # Create snapshot that will apply dynamic exclusions, and grab the set ID
+    RESPONSE=$( sudo curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snaps -X POST -H 'Content-Type: application/json' -d '{"action": "snapshot", "snaps": ["test-snap"], "snapshot-options": {"test-snap":{"exclude":["$SNAP_DATA/dynamic-exclude.txt", "$SNAP_COMMON/dynamic-exclude.txt", "$SNAP_USER_COMMON/dynamic-exclude.txt", "$SNAP_USER_DATA/dynamic-exclude.txt"]}}}' )
+    SET_ID=$( echo "$RESPONSE" | jq '.result."set-id"' )
+    echo "$SET_ID" | MATCH "^[0-9]+$"
+    CHANGE=$( echo "$RESPONSE" | jq ".change" | grep -o "[0-9]*" )
+    echo "$CHANGE" | MATCH "^[0-9]+$"
+    waitAsyncChangeDone "$CHANGE"
+
+    # Confirm the snapshot for the set ID includes the test-snap
+    snap saved --id="$SET_ID" | MATCH "test-snap"
+    # and is valid
+    snap check-snapshot "$SET_ID"
+    # and the snapshot metadata includes options
+    curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snapshots?set="$SET_ID" | MATCH '"options":'
+
+    # Check that shapshot is exportable
+    snap export-snapshot "$SET_ID" "${SET_ID}_export.snapshot"
+    test -e "${SET_ID}_export.snapshot"
+    # and importable
+    snap import-snapshot "${SET_ID}_export.snapshot"
+    # and the snapshot metadata includes the expected options entry (dynamic exclusions only)
+    cat <<EOF > expected_options_entry
+    "\$SNAP_DATA/dynamic-exclude.txt"
+    "\$SNAP_COMMON/dynamic-exclude.txt"
+    "\$SNAP_USER_COMMON/dynamic-exclude.txt"
+    "\$SNAP_USER_DATA/dynamic-exclude.txt"
+    EOF
+    curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snapshots?set="$SET_ID" | jq .result[].snapshots[].options.exclude[] > actual_options_entry
+    diff -u expected_options_entry actual_options_entry
+
+    # Remove the canaries to test restore
+    for DIR in "$USER_DATA_DIR" "$SYSTEM_DATA_DIR"
+    do
+        rm "$DIR/"{current,common}/canary.txt
+    done
+    
+    # Check restore
+    snap restore "$SET_ID" test-snap
+    MATCH "system versioned" < "$SYSTEM_DATA_DIR/current/canary.txt"
+    MATCH "system common" < "$SYSTEM_DATA_DIR/common/canary.txt"
+    MATCH "user versioned" < "$USER_DATA_DIR/current/canary.txt"
+    MATCH "user common" < "$USER_DATA_DIR/common/canary.txt"
+    cat <<EOF > expected_restored_files
+    /root/snap/test-snap
```

Comment:
you can use "$USER_DATA_DIR" and "$SYSTEM_DATA_DIR" so the paths are not harcoded

Reply:
Thanks, implemented